test/Tests/ClientTokenManagementTests.cs

```diff
@@ -30,6 +30,35 @@ async Task action()
         }
 
         await Should.ThrowAsync<InvalidOperationException>(action);
+    }
+
+    [Fact]
+    public async Task Missing_client_id_throw_exception()
+    {
+        var services = new ServiceCollection();
+
+        services.AddDistributedMemoryCache();
+        services.AddClientCredentialsTokenManagement()
+            .AddClient("test", client =>
+            {
+                client.TokenEndpoint = "https://as/connect/token";
+                client.ClientId = null;
+                client.ClientSecret = "client_secret";
+
+                client.Scope = "scope";
+                client.Resource = "resource";
+                client.Parameters.Add("audience", "audience");
+            });
+
+        var provider = services.BuildServiceProvider();
+        var sut = provider.GetRequiredService<IClientCredentialsTokenManagementService>();
+
+        async Task action()
+        {
+            var token = await sut.GetAccessTokenAsync("test");
+        }
+
+        await Should.ThrowAsync<InvalidOperationException>(action, "ClientId must not be empty");
```

Comment:
ThrowAsync returns the exception, so you can chain an assertion on to the end and assert that the message is what you expect.

Reply:
Fixed.